app/src/features/Batches/BatchCreate.tsx

```diff
@@ -88,16 +97,26 @@ const BatchCreate = (): JSX.Element => {
   };
 
   const handleBatchRun = async () => {
-    try {
-      await apiBatchCreate({
-        batchRequest: {
-          resources: selectedResourceIds,
-        },
-      }).unwrap();
-
-      setSelectedResourceIds([]);
-    } catch (e) {
-      setAlert(e.message as string);
+    refetchMappings();
+
+    if (mappings) {
```

Comment:
If we refetch mappings, will `mappings` value be the refreshed one right after this call ?